tests/integration/ha/test_backups.py

```diff
@@ -114,44 +119,201 @@ async def c_writes_runner(ops_test: OpsTest, c_writes: ContinuousWrites):
+    await http_request(
+        ops_test, "GET", f"https://{reachable_ip}:9200/_cat/shards", json_resp=False
+    )
+
     await c_writes.clear()
     logger.info("\n\n\n\nThe writes have been cleared.\n\n\n\n")
 
 
+@pytest.fixture(scope="session")
+def microceph():
+    """Starts microceph radosgw."""
+    import subprocess
+
+    if "microceph" not in subprocess.check_output(["sudo", "snap", "list"]).decode():
+        import os
+
+        import requests
+
+        uceph = "/tmp/microceph.sh"
+
+        with open(uceph, "w") as f:
+            resp = requests.get(
+                "https://raw.githubusercontent.com/canonical/microceph-action/main/microceph.sh"
+            )
+            f.write(resp.content.decode())
+
+        os.chmod(uceph, 0o755)
+        subprocess.check_output(
+            [
+                "sudo",
+                uceph,
+                "-c",
+                "latest/edge",
+                "-d",
+                "/dev/sdi",
+                "-a",
+                "accesskey",
+                "-s",
+                "secretkey",
+                "-b",
+                "data-charms-testing",
+                "-z",
+                "5G",
+            ]
+        )
+    # Now, return the configuration
+    ip = subprocess.check_output(["hostname", "-I"]).decode().split()[0]
+    return {"url": f"http://{ip}", "access-key": "accesskey", "secret-key": "secretkey"}
+
+
+async def _wait_backup_finish(ops_test, leader_id):
+    """Waits the backup to finish and move to the finished state or throws a RetryException."""
+    for attempt in Retrying(stop=stop_after_attempt(8), wait=wait_fixed(15)):
+        with attempt:
+            action = await run_action(
+                ops_test, leader_id, "list-backups", params={"output": "json"}
+            )
+            logger.debug(f"list-backups output: {action}")
+            # Expected format:
+            # namespace(status='completed', response={'return-code': 0, 'backups': '{"1": ...}'})
+            backups = json.loads(action.response["backups"])
+            logger.debug(f"Backups recovered: {backups}")
+            assert action.status == "completed"  # The actual action status
+            assert len(backups) > 0  # The number of backups
+            for backup in backups.values():
+                logger.debug(f"Backup is: {backup}")
+                assert backup["state"] == "SUCCESS"  # The backup status
+
+
+async def _assert_test_cont_writes_works(ops_test: OpsTest) -> ContinuousWrites:
+    """Asserts that TEST_BACKUP_INDEX is writable while under continuous writes.
+
+    Given we are restoring an index, we need to make sure ContinuousWrites restart at
+    the tip of that index instead of doc_id = 0.
+
+    Closes the writer at the end.
+    """
+    app = (await app_name(ops_test)) or APP_NAME
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    initial_count = await index_count(ops_test, app, leader_unit_ip, ContinuousWrites.INDEX_NAME)
```

Comment:
this should be `== 0` - as continuous writes are cleared after each test.

Reply:
As mentioned above, I am restoring the backup, it will contain a non-empty `series_index`. That is just to take an initial point of reference, then doing some other logic; finally checking if the count has increased.